rules/CodeQuality/Rector/Empty_/SimplifyEmptyCheckOnEmptyArrayRector.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodeQuality\Rector\Empty_;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\BinaryOp\Identical;
+use PhpParser\Node\Expr\Empty_;
+use PHPStan\Type\ArrayType;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\CodeQuality\Rector\Empty_\SimplifyEmptyCheckOnEmptyArrayRectorTest\SimplifyEmptyCheckOnEmptyArrayRectorTest
+ */
+final class SimplifyEmptyCheckOnEmptyArrayRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Simplify `empty` functions calls on empty arrays',
+            [new CodeSample('$array = []; if(empty($values))', '$array = []; if([] === $values)')]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Empty_::class];
+    }
+
+    /**
+     * @param Empty_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        // if empty($value) doesn't have array type return
+        if (! $this->nodeTypeResolver->getType($node->expr) instanceof ArrayType) {
+            return $node;
```

Comment:
return null; for no change:

```suggestion
            return null;
```

Reply:
++